Source/SwiftLintFramework/Rules/Lint/PeriodSpacingRule.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+import IDEUtils
+import SourceKittenFramework
+
+struct PeriodSpacingRule: SourceKitFreeRule, ConfigurationProviderRule, OptInRule, SubstitutionCorrectableRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "period_spacing",
+        name: "Period Spacing",
+        description: "Periods should not be followed by more than one space.",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("let pi = 3.2"),
+            Example("let pi = Double.pi"),
+            Example("let pi = Double. pi"),
+            Example("let pi = Double.  pi"),
+            Example("// A. Single."),
+            Example("///   - code: Identifier of the error. Integer."),
+            Example("""
+            // value: Multiline.
+            //        Comment.
+            """),
+            Example("""
+            /**
+            Sentence ended in period.
+
+            - Sentence 2 new line characters after.
+            **/
+            """)
+        ],
+        triggeringExamples: [
+            Example("/* Only god knows why.  This symbol does nothing. */", testWrappingInComment: false),
+            Example("// Only god knows why.  This symbol does nothing.", testWrappingInComment: false),
+            Example("// Single. Double.  End.", testWrappingInComment: false),
+            Example("// Single. Double.  Triple.   End.", testWrappingInComment: false),
+            Example("// Triple.   Quad.    End.", testWrappingInComment: false),
+            Example("///   - code: Identifier of the error.  Integer.", testWrappingInComment: false)
+        ],
+        corrections: [
+            Example("/* Why. ↓ Symbol does nothing. */"): Example("/* Why. Symbol does nothing. */"),
+            Example("// Why. ↓ Symbol does nothing."): Example("// Why. Symbol does nothing."),
+            Example("// Single. Double. ↓ End."): Example("// Single. Double. End."),
+            Example("// Single. Double. ↓ Triple. ↓  End."): Example("// Single. Double. Triple. End."),
+            Example("// Triple. ↓  Quad. ↓   End."): Example("// Triple. Quad. End."),
+            Example("///   - code: Identifier. ↓ Integer."): Example("///   - code: Identifier. Integer.")
+        ]
+    )
+
+    func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        // Find all comment tokens in the file and regex search them for violations
```

Comment:
I'd love to see a rule enforcing periods to finalize comments. 😉 Can we have a period after each sentence to be consistent?

Reply:
Uhhhh! I like that!
Should it be a separate rule though?

Reply:
Well, I don't think such a rule would be feasible. It's hard to recognize full sentences. A comment can also be a link, a code snippet or a few words which don't result into a sentence.

My comment was rather meant as a joke referring to your comments in the new rule ... 😄

Reply:
I'm glad that's the case, I had a mini crisis trying to figure out the rule for that 😅